src/file_storage.cpp

```diff
@@ -802,7 +802,7 @@ namespace libtorrent
 	std::int64_t file_storage::file_size(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		return std::int64_t(m_files[std::size_t(index)].size);
+		return m_files[std::size_t(index)].size;
```

Comment:
it's really unfortunate, I'm preparing a warning free branch and this is by far the biggest source of warnings. The problem is the conversion from signed to `std::size`.